gr-utils/modtool/tools/parser_cc_block.py

```diff
@@ -21,6 +21,27 @@ def dummy_translator(the_type, default_v=None):
     return the_type
 
 
+def remove_comments(inp: str) -> str:
+    def comment_removal(match: re.match) -> str:
+        groups = match.groups()
+        if len(groups) == 1:
+            return ''
+        if len(groups) % 3 == 0:
```

Comment:
What does this do?

Reply:
ah, good catch; that was from a more unified time, when I had a single regex that could rework `2 /* channels per polarizations */ * 2 /* polarizations */` into `2  * 2 ` by replacing a 

    (([not comment])*)((comment)*)(([not comment])*)

match by the not-comment parts. That was to enable nested comment starts, `1 /* /* comment */` being correctly removed, and to disable `"/*comment*"` in strings being removed, but then specialties of python regexes and having to implement the C escaping rules for quotes came in the way. I didn't want to maintain half a C++ tokenizer in a regex, so here I pareto'd and accepted that we'll be able to handle 
`3 /* max inputs */` and `2 /* channels per polarizations */ * 2 /* polarizations */`, but not the problem cases mentioned above. 


The group length check and 3-handling is a remnant of that, gimme a second.